src/applySchemaTyping.ts

```diff
@@ -0,0 +1,20 @@
+import type {LinkedJSONSchema} from './types/JSONSchema'
+import {typesOfSchema} from './typesOfSchema'
+
+export function applySchemaTyping(schema: LinkedJSONSchema) {
+  const types = typesOfSchema(schema)
+  schema.$types = types
+
+  if (types.length > 1) {
+    schema.$intersection = {
+      $id: schema.$id,
+      allOf: [],
+      description: schema.description,
+      title: schema.title,
+    }
+
+    delete schema.$id
+    delete schema.description
+    delete schema.name
```

Comment:
What about `title`?

Reply:
I missed `title` as it was not handled by the original [`maybeStripNameHints` function][1], though I have now added it.

[1]: https://github.com/bcherny/json-schema-to-typescript/blob/31993def993b610ba238d3024260129e31ddc371/src/utils.ts#L341-L352 'maybeStripNameHints'